Index: src/sord.c
===================================================================
--- src/sord.c	(revision 327)
+++ src/sord.c	(working copy)
@@ -148,6 +148,9 @@
 {
 	const SordNode* node = (const SordNode*)n;
 	uint32_t        hash = zix_digest_start();
+	if (node->node.buf == NULL) {
+		return 0;
+	}
 	hash = zix_digest_add(hash, node->node.buf, node->node.n_bytes);
 	hash = zix_digest_add(hash, &node->node.type, sizeof(node->node.type));
 	if (node->node.type == SERD_LITERAL) {
@@ -207,6 +210,7 @@
 		sord_node_free((SordWorld*)user_data, node->meta.lit.datatype);
 	}
 	free((uint8_t*)node->node.buf);
+	node->node.buf = NULL;
 }
 
 void
